Gems/ROS2/Code/Source/VehicleDynamics/DriveModel.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include "VehicleConfiguration.h"
+#include "VehicleInputsState.h"
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace VehicleDynamics
+{
+    //! Abstract class for turning vehicle inputs into behavior of wheels and steering elements
+    class DriveModel
+    {
+    public:
+        AZ_RTTI(DriveModel, "{1B57E83D-19BF-4403-8712-1AE98A12F0CD}");
+        enum DriveModelType
+        {
+            SimplifiedDriveModelType
+        };
```

Comment:
It is preferred to use a scoped enum here: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#enums
```suggestion
        enum class DriveModelType
        {
            SimplifiedDriveModelType
        };
```